### PR TITLE
#161348795 Enable Social login via Google, Facebook and Twitter

### PR DESCRIPTION
**Pull request by Sulaiman-Mozes**
#### What does this PR do?
Enable Social login via Google, Facebook and Twitter

#### Description of Task to be completed?
 - Users are able to login using Google, Facebook or Twitter
 - If the user is new, the user details are saved and a user is provided with a token
 - If the user is not new, a token is provide 
 - The username and email from the Oauth Provider are used to authenticate the user

#### How should this be manually tested?
 - By singing in using Google, Facebook or Twitter

#### What are the relevant pivotal tracker stories?
[#161348795]( https://www.pivotaltracker.com/story/show/161348795)

### CODE REVIEW

**Comment by brucemakallan** [comment]
`By signing in using Google, Facebook or Twitter` Please provide more detail on how this is done e.g. in Postman or Swagger

**Review by brucemakallan — CHANGES_REQUESTED** [review]
`By signing in using Google, Facebook or Twitter` Please provide more detail on how this is done e.g. in Postman or Swagger

File: authors/apps/authentication/urls.py [hunk 1]
```diff
@@ -2,17 +2,21 @@
 
 from .views import (
     LoginAPIView, RegistrationAPIView, AccountVerificationAPIView,
-    SendEmailPasswordReset, ResetPassword
+    SendEmailPasswordReset, ResetPassword, OauthAPIView, OauthlLoginAPIView
 )
 
 urlpatterns = [
-    # path('users/<pk>', UserRetrieveUpdateAPIView.as_view()), Story not yet worked on
+    # path('users/<pk>', UserRetrieveUpdateAPIView.as_view()),
```
[the remark this answers is not in the document]
Comment (emugaya) [reply authors/apps/authentication/urls.py]: @Sulaiman-Mozes Fix this !!!!

File: authors/apps/authentication/migrations/0001_initial.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 2.1.2 on 2018-11-05 08:37
+# Generated by Django 2.1.2 on 2018-11-07 04:36
```
Comment (emugaya) [inline authors/apps/authentication/migrations/0001_initial.py:1]: Why are you updating this?

File: authors/apps/authentication/urls.py (hunk 1 above)
Comment (emugaya) [inline authors/apps/authentication/urls.py]: Remove commented code

File: authors/apps/authentication/views.py [hunk 3]
```diff
@@ -199,3 +232,42 @@ def put(self, request, *args, **kwargs):
...
+
+
+class OauthAPIView(generics.GenericAPIView):
+    permission_classes = (AllowAny,)
+    # renderer_classes = (UserJSONRenderer,)
+
+    def get(self, request, social_auth_Provider, auth_provider_url=None):
+        social_auth_Provider = social_auth_Provider.lower()
+
+        if social_auth_Provider == 'google':
```
Comment (emugaya) [inline authors/apps/authentication/views.py:252]: Refactor line 244 to 252 into one method

File: authors/apps/authentication/migrations/0001_initial.py (hunk 2 above)
Comment (Sulaiman-Mozes) [reply authors/apps/authentication/migrations/0001_initial.py:1]: It is automatically updated when you run migrations

**Pull request merged**